runtime/sema/type.go

```diff
@@ -6448,6 +6518,33 @@ func (t *AddressType) initializeMemberResolvers() {
 	})
 }
 
+func IsPrimitiveOrContainerOfPrimitive(ty Type) bool {
+	if ty.IsPrimitiveType() {
+		return true
+	}
+
+	// TODO: Do we also want to count Dictionary?
+	switch ty.(type) {
```

Comment:
```suggestion
	switch typedTy := ty.(type) {
```

Reply:
This way you don't have to manually cast inside the `case`s below, or panic on an unreachable failure.

Reply:
Or even shadow: `ty := ty.(type)`

Reply:
thanks, done.